alicloud/resource_alicloud_emr_cluster.go

```diff
@@ -0,0 +1,551 @@
+
+		oldHostGroup := map[string]map[string]interface{}{}
+		for _, v := range v1.(*schema.Set).List() {
+			hostGroupName := v.(map[string]interface{})["host_group_name"].(string)
+			oldHostGroup[hostGroupName] = v.(map[string]interface{})
+		}
+
+		newHostGroup := map[string]map[string]interface{}{}
+		for _, v := range v2.(*schema.Set).List() {
+			hostGroupName := v.(map[string]interface{})["host_group_name"].(string)
+			newHostGroup[hostGroupName] = v.(map[string]interface{})
+		}
+
+		resizeRequest := emr.CreateResizeClusterV2Request()
+		resizeRequest.ClusterId = d.Id()
+
+		var resizeHostGroups []emr.ResizeClusterV2HostGroup
+
+		releaseRequest := emr.CreateReleaseClusterHostGroupRequest()
+		releaseRequest.ClusterId = d.Id()
+
+		for k, v1 := range newHostGroup {
+			if _, ok := oldHostGroup[k]; ok {
+				newNodeCount, _ := strconv.Atoi(v1["node_count"].(string))
+
+				listHostGroupRequest := emr.CreateListClusterHostGroupRequest()
+				listHostGroupRequest.ClusterId = d.Id()
+				listHostGroupRequest.HostGroupName = k
+
+				raw, err := client.WithEmrClient(func(emrClient *emr.Client) (interface{}, error) {
+					return emrClient.ListClusterHostGroup(listHostGroupRequest)
+				})
+				if err != nil {
+					return WrapErrorf(err, DefaultErrorMsg, "alicloud_emr_cluster", listHostGroupRequest.GetActionName(), AlibabaCloudSdkGoERROR)
+				}
+
+				resp := raw.(*emr.ListClusterHostGroupResponse)
+				hostGroupId := resp.HostGroupList.HostGroup[0].HostGroupId
+				oldNodeCount := resp.HostGroupList.HostGroup[0].NodeCount
+
+				// scala up
+				if oldNodeCount < newNodeCount {
+					count := newNodeCount - oldNodeCount
+					resizeHostGroup := emr.ResizeClusterV2HostGroup{}
+					resizeHostGroup.ClusterId = d.Id()
+					resizeHostGroup.HostGroupId = hostGroupId
+					resizeHostGroup.NodeCount = strconv.Itoa(count)
+					resizeHostGroup.InstanceType = v1["instance_type"].(string)
+					resizeHostGroup.HostGroupType = v1["host_group_type"].(string)
+
+					resizeHostGroups = append(resizeHostGroups, resizeHostGroup)
+				} else if oldNodeCount > newNodeCount { //scale down
+					releaseRequest.HostGroupId = hostGroupId
+					releaseRequest.InstanceIdList = v1["instance_list"].(string)
+
+					_, err := client.WithEmrClient(func(emrClient *emr.Client) (interface{}, error) {
+						return emrClient.ReleaseClusterHostGroup(releaseRequest)
+					})
+
+					if err != nil {
+						return WrapErrorf(err, DefaultErrorMsg, "alicloud_emr_cluster", releaseRequest.GetActionName(), AlibabaCloudSdkGoERROR)
+					}
+				}
+			}
+		}
+
+		if len(resizeHostGroups) != 0 {
+			resizeRequest.HostGroup = &resizeHostGroups
+
+			_, err := client.WithEmrClient(func(emrClient *emr.Client) (interface{}, error) {
+				return emrClient.ResizeClusterV2(resizeRequest)
+			})
+
+			if err != nil {
+				return WrapErrorf(err, DefaultErrorMsg, "alicloud_emr_cluster", resizeRequest.GetActionName(), AlibabaCloudSdkGoERROR)
+			}
+		}
+		d.SetPartial("host_group")
```

Comment:
Does the cluster status change when changing `host_group`? If so, please wait its status to target status after that.

Reply:
the status does not change

Reply:
If not, please remove `Update: schema.DefaultTimeout(10 * time.Minute),`